validator/lib/validateBoundingVolume.js

```diff
@@ -0,0 +1,109 @@
+'use strict';
+
+var Promise = require('bluebird');
+var Cesium = require('cesium');
+var defined = Cesium.defined;
+
+module.exports = validateVolume;
+
+function validateVolume(tileset) {
+    return new Promise(function(resolve) {
+        validateNode(tileset.root, tileset, resolve);
+    });
+}
+
+var scratchCartographic = new Cartographic();
+var scratchRect = new Rectangle();
+var scratchTileRect = new Rectangle();
+
+function regionInsideRegion() {
+    return (Rectangle.contains(scratchTileRect, Cartographic.fromRadians(scratchRect.west,
+        scratchRect.north, 0.0, scratchCartographic))
+    && Rectangle.contains(scratchTileRect, Cartographic.fromRadians(scratchRect.west,
+        scratchRect.south, 0.0, scratchCartographic))
+    && Rectangle.contains(scratchTileRect, Cartographic.fromRadians(scratchRect.east,
+        scratchRect.north, 0.0, scratchCartographic))
+    && Rectangle.contains(scratchTileRect, Cartographic.fromRadians(scratchRect.east,
+        scratchRect.south, 0.0, scratchCartographic)));
+}
+
+function sphereInsideSphere(sphere, tileSphere) {
+  var sphereX = sphere[0];
+  var sphereY = sphere[1];
+  var sphereZ = sphere[2];
+  var sphereRadius = sphere[3];
+  var tileSphereX = tileSphere[0];
+  var tileSphereY = tileSphere[1];
+  var tileSphereZ = tileSphere[2];
+  var distance = (sphereX - tileSphereX)*(sphereX - tileSphereX) + (sphereY - tileSphereY)*(sphereY - tileSphereY) +
+      (sphereZ - tileSphereZ)*(sphereZ - tileSphereZ);
+  var tileRadius = tileSphere[3];
+
+  return (distance <= sphereRadius^2 && tileRadius <= sphereRadius);
+
+}
+
+function validateNode(root, parent, resolve) {
+    var stack = [];
+    stack.push({
+        node: root,
+        parent: parent
+    });
+
+    while (stack.length > 0) {
+        var node = stack.pop();
+        var tile = node.node;
+        var parent = node.parent;
+
+        if (defined(tile.content)) {
+            if (defined(tile.content.boundingVolume)) {
```

Comment:
These two lines can be combined. Factor out `tile.content` to its own variable to avoid repeating it a lot.